src/query/src/promql/planner.rs

```diff
@@ -941,9 +952,15 @@ impl PromPlanner {
     }
 
     // TODO(ruihang): ignore `MetricNameLabel` (`__name__`) matcher
-    fn matchers_to_expr(&self, label_matchers: Matchers) -> Result<Vec<DfExpr>> {
+    fn matchers_to_expr<F>(label_matchers: Matchers, filter: F) -> Result<Vec<DfExpr>>
+    where
+        F: Fn(&str) -> bool,
+    {
         let mut exprs = Vec::with_capacity(label_matchers.matchers.len());
         for matcher in label_matchers.matchers {
+            if !filter(&matcher.name) {
```

Comment:
Document the behaviour.

Reply:
resolved in [40d5547](https://github.com/GreptimeTeam/greptimedb/pull/5519/commits/40d55472abeba66ef4ddd918862abac5793da6bf)